test/allocators/LobiAllocator.test.ts

```diff
@@ -0,0 +1,452 @@
+                loss: bne(10, 8),
+            });
+
+            await allocator.activate();
+
+            const amount: BigNumber = bne(10, 9);
+
+            await expect(() => extender.requestFundsFromTreasury(1, amount)).to.changeTokenBalance(
+                lobi,
+                allocator,
+                amount
+            );
+
+            await expect(() => allocator.update(1)).to.changeTokenBalance(
+                lobi,
+                allocator,
+                BigNumber.from(0).sub(amount)
+            );
+        });
+
+        it("should revert if sender is not guardian", async () => {
+            await expect(allocator.connect(owner).deallocate([1])).to.be.reverted;
+        });
+
+        it("should fully deallocate", async () => {
+            for (let i = 0; i < 1000; i++) {
+                await triggerRebase();
+            }
+
+            const balance = await utilTokens[0].balanceOf(allocator.address);
+            const input: BigNumber[] = [balance];
+            await allocator.deallocate(input);
+
+            expect(await utilTokens[0].balanceOf(allocator.address)).to.equal("0");
+            expect(await tokens[0].balanceOf(allocator.address)).to.equal(balance);
+        });
+
+        it("should partially deallocate", async () => {
+            const balance = await utilTokens[0].balanceOf(allocator.address);
+            const input: BigNumber[] = [balance.div("2")];
+            await allocator.deallocate(input);
+
+            expect(await utilTokens[0].balanceOf(allocator.address)).to.equal(balance.div("2"));
+            expect(await tokens[0].balanceOf(allocator.address)).to.equal(balance.div("2"));
+        });
+    });
+
+    describe("migrate()", () => {
+        beforeEach(async () => {
+            allocator = await factory.deploy(
+                {
+                    authority: authority.address,
+                    tokens: [lobi.address],
+                    extender: extender.address,
+                },
+                treasury.address,
+                sLobi.address,
+                "0x3818eff63418e0a0BA3980ABA5fF388b029b6d90",
+                "0x644D94dA13af3ac88a9a0dcaaA108E474B9a9B5F"
+            );
+
+            await extender.registerDeposit(allocator.address);
+
+            await extender.setAllocatorLimits(1, {
+                allocated: bne(10, 20),
+                loss: bne(10, 8),
+            });
+
+            await allocator.activate();
+
+            const amount: BigNumber = bne(10, 9);
+
+            await expect(() => extender.requestFundsFromTreasury(1, amount)).to.changeTokenBalance(
+                lobi,
+                allocator,
+                amount
+            );
+
+            await expect(() => allocator.update(1)).to.changeTokenBalance(
+                lobi,
+                allocator,
+                BigNumber.from(0).sub(amount)
+            );
+
+            const mAllocator: LobiAllocator = await factory.deploy(
```

Comment:
if this deploy is being done a lot then just make this into a function you call.